src/python/pants/core/util_rules/archive.py

```diff
@@ -79,31 +98,91 @@ async def find_tar() -> TarBinary:
     return TarBinary(first_path.path, first_path.fingerprint)
 
 
+# -----------------------------------------------------------------------------------------------
+# Create archives
+# -----------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class CreateArchive:
+    """A request to create an archive.
+
+    All files in the input snapshot will be included in the resulting archive.
+    """
+
+    snapshot: Snapshot
+    output_filename: str
+    format: ArchiveFormat
+
+
+@rule(desc="Creating an archive file", level=LogLevel.DEBUG)
+async def create_archive(
+    request: CreateArchive, tar_binary: TarBinary, zip_binary: ZipBinary
+) -> Digest:
+    if request.format == ArchiveFormat.ZIP:
+        argv = zip_binary.create_archive_argv(request)
+        env = {}
+        input_digest = request.snapshot.digest
+    else:
+        argv = tar_binary.create_archive_argv(request)
+        # `tar` expects to find a couple binaries like `gzip` and `xz` by looking on the PATH.
+        env = {"PATH": os.pathsep.join(SEARCH_PATHS)}
```

Comment:
Can we have an option (maybe as a follow-on) to reply on the remote image's PATH?

Reply:
What do you mean? Like the user can specify the PATH similar to `--pex-ecutable-search-paths`? There's a TODO above about that, if this should be configurable.